tsdb/docs/format/wal.md

```diff
@@ -17,7 +17,8 @@ Notable deviations are that the record fragment is encoded as:
 └───────────┴──────────┴────────────┴──────────────┘
 ```
 
-The type flag has the following states:
+The type flag is comprised of three components: a 3-bit reserved field, a 1-bit zstd compression flag, a 1-bit snappy compression flag, and a 3-bit type flag. 
```

Comment:
"flag" doesn't seem very meaningful here; it appears at the beginning and end of the line for different things.

And "is comprised of" is often considered an error: https://en.wikipedia.org/wiki/Comprised_of

How about "The initial type byte is made up of ..."

Also we should probably say where these bitfields appear in the byte.

Reply:
How about "The initial header byte is made up of ..."